docs/proposals/observability.md

```diff
@@ -0,0 +1,311 @@
+	// TargetRef identifies an API object to apply the policy to.
+	// Object must be in the same namespace as the policy.
+	// Support: HTTPRoute and HTTPRoute rule
+	TargetRef gatewayv1alpha2.PolicyTargetReferenceWithSectionName `json:"targetRef"`
+
+	// Tracing allows for enabling and configuring tracing.
+    //
+	// +optional
+	Tracing *Tracing `json:"tracing,omitempty"`
+}
+
+// Tracing allows for enabling and configuring OpenTelemetry tracing.
+type Tracing struct {
+	// Ratio is the percentage of traffic that should be sampled. Integer from 0 to 100.
+	// By default, 100% of http requests are traced. Not applicable for parent-based tracing.
+    //
+	// +optional
+	Ratio *int32 `json:"ratio,omitempty"`
+
+	// Context specifies how to propagate traceparent/tracestate headers. By default is 'ignore'.
+    //
+	// +optional
+	Context *TraceContext `json:"context,omitempty"`
+
+	// SpanName defines the name of the Otel span. By default is the name of the location for a request.
+    //
+	// +optional
+	SpanName *string `json:"spanName,omitempty"`
+
+	// SpanAttributes are custom key/value attributes that are added to each span.
+    //
+	// +optional
+	SpanAttributes map[string]string `json:"spanAttributes,omitempty"`
+
+	// Enable defines if tracing is enabled, disabled, or parent-based.
+	Enable TraceType `json:"enable"`
+}
+
+// TraceType defines if tracing is enabled.
+type TraceType string
+
+const (
+	// TraceTypeOn enables tracing.
+	TraceTypeOn TraceType = "on"
+
+	// TraceTypeOff disables tracing.
+	TraceTypeOff TraceType = "off"
+
+	// TraceTypeParent enables tracing and only records spans if the parent span was sampled.
+	TraceTypeParent TraceType = "parent"
+)
+
+// TraceContext specifies how to propagate traceparent/tracestate headers.
+type TraceContext string
+
+const (
+	// TraceContextExtract uses an existing trace context from the request, so that the identifiers
+	// of a trace and the parent span are inherited from the incoming request.
+	TraceContextExtract TraceContext = "extract"
+
+	// TraceContextInject adds a new context to the request, overwriting existing headers, if any.
+	TraceContextInject TraceContext = "inject"
+
+	// TraceContextPropagate updates the existing context (combines extract and inject).
+	TraceContextPropagate TraceContext = "propagate"
+
+	// TraceContextIgnore skips context headers processing.
+	TraceContextIgnore TraceContext = "ignore"
+)
+```
+
+### YAML
+
+Below is an example YAML version of an `ObservabilityPolicy`:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: ObservabilityPolicy
+metadata:
+  name: example-observability-policy
+  namespace: default
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: example-route
+    sectionName: example-section
+  tracing:
+    ratio: 10
+    context: inject
+    spanName: example-span
+    spanAttributes:
+      attribute1: value1
+      attribute2: value2
+    enable: on
```

Comment:
on is a boolean value. to pass it as a string, we need to put it in quotes. "on".
this will probably a common mistake among users.